File: src/com/dafttech/terra/engine/input/FocusManager.java
package com.dafttech.terra.engine.input;

public class FocusManager {
    static IFocusableTyping typeFocus = null;
    
    public static boolean acquireTypeFocus(IFocusableTyping listener) {
        if(typeFocus != null) return false;
        typeFocus = listener;
        return true;
    }
    
    public static boolean releaseTypeFocus(IFocusableTyping listener) {
        if(typeFocus != listener) return false;
        typeFocus = null;
        return true;
    }
    
    public static boolean typeFocusAssigned() {
        return typeFocus != null;
    }
}

File: src/main/java/com/dafttech/terra/engine/input/FocusManager.scala
package com.dafttech.terra.engine.input

object FocusManager {
  private[input] var typeFocus: IFocusableTyping = null

  def acquireTypeFocus(listener: IFocusableTyping): Boolean = {
    if (typeFocus != null) return false
    typeFocus = listener
    return true
  }

  def releaseTypeFocus(listener: IFocusableTyping): Boolean = {
    if (typeFocus ne listener) return false
    typeFocus = null
    return true
  }

  def hasTypeFocus(listener: IFocusableTyping): Boolean = {
    return typeFocus eq listener
  }

  def typeFocusAssigned: Boolean = {
    return typeFocus != null
  }
}